internal/service/streaming_processor.go

```diff
@@ -0,0 +1,506 @@
+package service
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/gammazero/workerpool"
+	"github.com/hashicorp/go-retryablehttp"
+
+	"github.com/flexprice/flexprice/internal/domain/task"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/httpclient"
+	"github.com/flexprice/flexprice/internal/logger"
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// StreamingProcessor handles streaming processing of large files
+type StreamingProcessor struct {
+	Client           httpclient.Client
+	Logger           *logger.Logger
+	ProviderRegistry *FileProviderRegistry
+	CSVProcessor     *CSVProcessor
+	JSONProcessor    *JSONProcessor
+	WorkerPool       *workerpool.WorkerPool
```

Comment:
I don't think we need to this as it is not getting used anywhere so can clean up in follow up PR